### PR TITLE
only use dcm files for export

### PR DESCRIPTION
**Pull request by jackschofield23**
Signed-off-by: Jack Schofield <jack.schofield@answerdigital.com>

### Description

Fixes # .

Only submits dicom files for export.
Now fails task and workflow if there are no files to export.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### CODE REVIEW

File: src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 1]
```diff
@@ -365,15 +358,29 @@ private async Task HandleDicomExportAsync(WorkflowRevision workflow, WorkflowIns
...
+                        files.AddRange(dcmFiles.ToList());
                     }
                 }
             }
 
             artifactValues = files.Select(f => f.FilePath).ToArray();
 
+            if (artifactValues.IsNullOrEmpty())
+            {
+                _logger.ExportFilesNotFound(task.TaskId, workflowInstance.Id);
```
Comment (neildsouth) [inline src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs:375]: This should probably be at least at Warning Log Level, maybe even Error !

Comment (jackschofield23) [reply src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs:375]: Changed this to be Error.

**Review by neildsouth — APPROVED** [review]

File: src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 2]
```diff
@@ -365,15 +358,29 @@ private async Task HandleDicomExportAsync(WorkflowRevision workflow, WorkflowIns
                         workflowInstance.BucketId,
                         artifact,
                         true);
-                    if (objects.IsNullOrEmpty() is false)
+
+                    var dcmFiles = objects?.Where(o => o.FilePath.EndsWith(".dcm"))?.ToList();
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs:362]: Does this change also removes the ability to export HL7/FHIR?

[the remark this answers is not in the document]
Comment (jackschofield23) [reply src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs:362]: Do we know what other file extensions we will support?

[the remark this answers is not in the document]
Comment (jackschofield23) [reply src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs:362]: Currently we only support exporting to the dicom scu endpoint. So when we extend the export to other endpoints we can update supported filetypes.

[the remark this answers is not in the document]
Comment (mocsharp) [reply src/WorkflowManager/WorkflowExecuter/Services/WorkflowExecuterService.cs:362]: We will need to support DICOM, FHIR, and HL7 but no guarantee what file extensions users will use. Therefore, I think we must document the usage, e.g., without `.dcm` we will not export their DICOM files.

**Pull request merged**